plugins/pycv/PlumedPythonEmbeddedModule.cpp

```diff
@@ -0,0 +1,273 @@
+       "returns an interface to the current pbcs")
+  .def("getNeighbourList",&PLMD::pycv::PythonCVInterface::getNL,
+       "returns an interface to the current Neighborlist")
+
+//https://pybind11.readthedocs.io/en/stable/advanced/functions.html#return-value-policies
+  /*.def("getAbsoluteIndexes", &PLMD::pycv::PythonCVInterface::getAbsoluteIndexes ,
+  "Get the vector of absolute indexes.",
+  py::return_value_policy::reference)*/
+  //using unsigned: if AtomNumber changes, also this must change
+  .def("absoluteIndexes",[](PLMD::pycv::PythonCVInterface* self) -> py::array_t<unsigned> {
+    auto nat=self->getPositions().size();
+    py::array_t<unsigned>::ShapeContainer shape({nat});
+    py::array_t<unsigned> atomIndexes(shape);
+    auto retAccessor = atomIndexes.mutable_unchecked<1>();
+    for (decltype(nat) i=0; i < nat; ++i) {
+      //at time of writing getAbsoluteIndexes returns const std::vector<AtomNumber> &
+      retAccessor(i)=self->getAbsoluteIndexes()[i].index();
+    }
+    return atomIndexes;
+  },
+  "Get the vector of absolute indexes.")
+
+  .def("charge", &PLMD::pycv::PythonCVInterface::getCharge,
+       "Get charge of i-th atom", py::arg("i"))
+
+  .def("mass", &PLMD::pycv::PythonCVInterface::getMass,
+       "Get mass of i-th atom", py::arg("i"))
+
+  .def("masses",
+  [](PLMD::pycv::PythonCVInterface* self) -> py::array_t<double> {
+    auto nat=self->getPositions().size();
+    py::array_t<double>::ShapeContainer shape({nat});
+    py::array_t<double> masses(shape);
+    auto retAccessor = masses.mutable_unchecked<1>();
+    for (decltype(nat) i=0; i < nat; ++i) {
+      retAccessor(i)=self->getMass(i);
+    }
+    return masses;
+  },
+  "Returns and ndarray with the masses of the atoms requested by the action")
+
+  .def("charges",
+  [](PLMD::pycv::PythonCVInterface* self) -> py::array_t<double> {
+    auto nat=self->getPositions().size();
+    py::array_t<double>::ShapeContainer shape({nat});
+    py::array_t<double> charges(shape);
+    auto retAccessor = charges.mutable_unchecked<1>();
+    for (decltype(nat) i=0; i < nat; ++i) {
+      retAccessor(i)=self->getCharge(i);
+    }
+    return charges;
+  },
+  "Returns and ndarray with the charges of the atoms requested by the action");
+
+  /***********************************PLMD::Pbc********************************/
+  py::class_<PLMD::Pbc>(m, "Pbc")
+  //.def(py::init<>())
+  .def("apply",[](const PLMD::Pbc* self, py::array_t<double>& deltas) -> py::array_t<double> {
+    //TODO:shape check
+    //TODO: this may be set up to modify the passed deltas, so no new intialization
+    //      ^this needs to be VERY clear to te user
+    auto accessor = deltas.unchecked<2>();
+    auto nat=deltas.shape(0);
+    py::array_t<double> toRet({nat,deltas.shape(1)});
+    auto retAccessor = toRet.mutable_unchecked<2>();
+    for (decltype(nat) i=0; i < nat; ++i) {
+      auto t= self->distance(PLMD::Vector(0.0,0.0,0.0),
```

Comment:
@Iximiel I think it might be useful to add this functionality to the Pbc class itself, I mean: the possibility to operate (probably in-place) on a `(PLMD::Vector*vectors,std::size_t size)` or on a `std::vector<PLMD::Vector>`. I guess  it would be more efficient (but it can be done later)